core/thread/id.go

```diff
@@ -184,84 +171,138 @@ func uvError(read int) error {
 	}
 }
 
+func (i ID) Valid() bool {
```

Comment:
Made this return `bool`, but I'm thinking `func (i ID) Validate() error` might be cleaner to use. Opinion?

Reply:
Yeah, I guess error would allow for finer-grained feedback